runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/types/KvCoderComparator.java

```diff
@@ -78,10 +78,12 @@ private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundE
   @Override
   public int hash(WindowedValue<KV<K, V>> record) {
     K key = record.getValue().getKey();
-    if (key != null) {
-      return key.hashCode();
-    } else {
-      return 0;
+    buffer1.reset();
+    try {
+      keyCoder.encode(key, buffer1, Coder.Context.OUTER);
```

Comment:
If you're willing to allocate some new objects, this reset + try block can be reduced to `return Arrays.hash(CoderUtils.encodeToByteArray(keyCoder, key))`